lumicks/pylake/kymo.py

```diff
@@ -269,6 +272,54 @@ def plot_with_time_histogram(self, color_channel, pixels_per_bin=1, hist_ratio=0
         ax_hist.set_ylabel("counts")
         ax_hist.set_title(self.name)
 
+    def crop_by_distance(self, lower, upper):
+        """Crop the kymo by position.
+
+        Crops the kymograph down to lower <= x < upper.
+
+        Parameters
+        ----------
+        lower : float
+            Lower bound in physical units.
+        upper : float
+            Upper bound in physical units.
+        """
+        lower_pixels = int(lower / self.pixelsize_um[0])
+        upper_pixels = int(np.ceil(upper / self.pixelsize_um[0]))
+        if lower_pixels < 0 or upper_pixels < 0:
+            raise ValueError("Cropping by negative positions not allowed")
+
+        n_pixels = len(np.arange(self._num_pixels[0])[lower_pixels:upper_pixels])
+        if n_pixels == 0:
+            raise IndexError("Cropped image would be empty")
+
+        position_offset = self.position_offset + lower_pixels * self.pixelsize_um[0]
+        result = Kymo(self.name, self.file, self.start, self.stop, self._json, position_offset)
+
+        def image_factory(_, channel):
+            return self._image(channel)[lower_pixels:upper_pixels, :]
+
+        def timestamp_factory(_, reduce):
+            return self._timestamps("timestamps", reduce)[lower_pixels:upper_pixels, :]
+
+        def line_time_factory(_):
+            return self.line_time_seconds
+
+        def pixelsize_factory(_):
+            return self.pixelsize_um
```

Comment:
I think for things like this where you just forward the existing factory it might be cleaner to just have these as lambdas down below. It could also make a clear distinction between things that are changed or not if we consistently use lambdas for forwarded values and function definitions for changed

Reply:
Sure. I'll update it to do that :)